src/backend/services/wallet-api/src/routes.py

```diff
@@ -31,6 +36,40 @@ def getWalletAmount():
     response = send_get_request( blockchain_wallet_url.format("balance"), data)
     return jsonify(response.json()), response.status_code
 
+@cross_origin()
+@app.route("/history", methods=['GET'])
+def getWalletHistory():
+
+    umnetId = request.args.get('umnetId')
+
+    if umnetId is None:
+        raise IncorrectPayloadException()
+
+    # response = requests.get("http://localhost/blockchain/chain").json()
+    response = requests.get( blockchain_url.format("chain")).json()
+    chain = response['chain']
+
+    result = []
+
+    for block in chain[1:]:
+        block = json.loads(block)
+        ts = float(block["transaction"]["timestamp"])
+        
+        # if user was the miner 
+        if umnetId.upper() == block["miner_id"].upper() :
```

Comment:
More efficient to just do `umnetId = umnetId.upper()` once above instead of in every one of these conditionals

Reply:
+1 on this

Reply:
agreed, done